src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl

```diff
@@ -132,55 +167,128 @@ function SplitExplicitAuxiliary(grid::AbstractGrid)
     dz = GridMetricOperation((Center, Center, Center), Δz, grid)
     sum!(Hᶜᶜ, dz)
 
-    return SplitExplicitAuxiliary(; Gᵁ, Gⱽ, Hᶠᶜ, Hᶜᶠ, Hᶜᶜ)
+    fill_halo_regions!((Hᶠᶜ, Hᶜᶠ, Hᶜᶜ))
+
+    kernel_size    = :xy
+    kernel_offsets = (0, 0)
+
+    return SplitExplicitAuxiliary(; Gᵁ, Gⱽ, Hᶠᶜ, Hᶜᶠ, Hᶜᶜ, kernel_size, kernel_offsets)
 end
 
 """
-    struct SplitExplicitSettings{𝒩, ℳ}
+    struct SplitExplicitSettings
 
 A struct containing settings for the split-explicit free surface.
 
 $(TYPEDFIELDS)
 """
-struct SplitExplicitSettings{𝒩, ℳ}
+struct SplitExplicitSettings{𝒩, ℳ, 𝒯, 𝒮}
     "substeps: (`Int`)"
     substeps :: 𝒩
-    "velocity_weights : (`Vector`)"
-    velocity_weights :: ℳ
-    "free_surface_weights : (`Vector`)"
-    free_surface_weights :: ℳ
+    "averaging_weights : (`Vector`)"
+    averaging_weights :: ℳ
+    "mass_flux_weights : (`Vector`)"
+    mass_flux_weights :: ℳ
+    "fractional step: (`Number`), the barotropic time step will be (Δτ ⋅ Δt)" 
+    Δτ :: 𝒯
+    "time stepping scheme"
+    timestepper :: 𝒮
 end
 
-function SplitExplicitSettings(; substeps = 200, velocity_weights = nothing, free_surface_weights = nothing)
-    velocity_weights = Tuple(ones(substeps) ./ substeps)
-    free_surface_weights = Tuple(ones(substeps) ./ substeps)
+"""
+    Possible barotropic time-stepping scheme. 
 
-    return SplitExplicitSettings(substeps,
-        velocity_weights,
-        free_surface_weights)
+- `AdamsBashforth3Scheme`: η = f(U, Uᵐ⁻¹, Uᵐ⁻²) then U = f(η, ηᵐ, ηᵐ⁻¹, ηᵐ⁻²)
+- `ForwardBackwardScheme`: η = f(U)             then U = f(η)
+"""
+struct AdamsBashforth3Scheme end
+struct ForwardBackwardScheme end
```

Comment:
Maybe should make these specific to split explicit so we don't confuse with "other" time stepping schemes -- helps when using `grep` to search the code base (or reviewing PRs)